src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2500,6 +2500,18 @@ void ApplicationManagerImpl::UnregisterApplication(
                             << is_resuming
                             << "; is_unexpected_disconnect = "
                             << is_unexpected_disconnect);
+  bool send_unsubscribed_way_points;
```

Comment:
@OHerasym you can just do next way:

```
size_t size = 0;
{
    sync_primitives::AutoLock lock(subscribed_way_points_apps_lock_);
    size = subscribed_way_points_apps_list_.size();
}
if (1 == size) {
    LOG4CXX_ERROR(logger_, "Send UnsubscribeWayPoints");
    MessageHelper::SendUnsubscribedWayPoints();
}
```

I think it would be better